src/rdkafka_request.c

```diff
@@ -1560,8 +1560,8 @@ void rd_kafka_ApiVersionRequest (rd_kafka_broker_t *rkb,
 	/* 0.9.0.x brokers will not close the connection on unsupported
 	 * API requests, so we minimize the timeout to 10s for the request.
 	 * This is a regression on the broker part. */
-	if (rkb->rkb_rk->rk_conf.socket_timeout_ms > 10*1000)
-		rkbuf->rkbuf_ts_timeout = rd_clock() + (10 * 1000 * 1000);
+	if (rkb->rkb_rk->rk_conf.socket_timeout_ms > rkb->rkb_rk->rk_conf.api_version_request_timeout_ms)
```

Comment:
I don't think we need this check anymore, it was only there to make a low socket.timeout.ms take precedence over the hard-coded 10s timeout, but with a specific config property we can just set rkbuf_ts_timeout to the api_version_request_timeout.ms.